### PR TITLE
Adding support for header

### PR DESCRIPTION
**Pull request by Jerem584**
Hi,

so... 

1. The code in my repo is very messy
2. Code work fine, Tests & examples updated also

### CODE REVIEW

File: test/http_client.cpp [hunk 1]
```diff
@@ -6,7 +6,8 @@
 TEST(siesta, client_connect_ok)
 {
     auto f = siesta::client::getRequest("http://postman-echo.com/get?foo1=bar1",
-                                        5000);
+        std::vector<std::tuple<std::string, std::string>>(),
```
Comment (robiwano) [inline test/http_client.cpp:9]: Since a header item is a key+value pair, I'd recommend using std::pair instead of std::tuple for clarity.

Comment (Jerem584) [reply test/http_client.cpp]: Ok i updated (see newer commit)

improved exemples & src code

**Comment by Jerem584** [comment]
any news?

**Comment by robiwano** [comment]
Sure. Could you please squash the branch into one commit?

**Comment by Jerem584** [comment]
done

**Comment by robiwano** [comment]
Thanks for your contribution!

**Pull request merged**